test/memoryPoolAPI.cpp

```diff
@@ -51,48 +64,40 @@ TEST_F(test, memoryPoolTrace) {
     ASSERT_EQ(providerCalls["alloc"], 1);
     ASSERT_EQ(providerCalls.size(), ++provider_call_count);
 
+    umfPoolMallocUsableSize(tracingPool.get(), nullptr);
+    ASSERT_EQ(poolCalls["malloc_usable_size"], 1);
+    ASSERT_EQ(poolCalls.size(), ++pool_call_count);
+
+    ASSERT_EQ(providerCalls.size(), provider_call_count);
+
     umfPoolFree(tracingPool.get(), nullptr);
     ASSERT_EQ(poolCalls["free"], 1);
     ASSERT_EQ(poolCalls.size(), ++pool_call_count);
 
     ASSERT_EQ(providerCalls["free"], 1);
     ASSERT_EQ(providerCalls.size(), ++provider_call_count);
 
-    umfPoolCalloc(tracingPool.get(), 0, 0);
-    ASSERT_EQ(poolCalls["calloc"], 1);
-    ASSERT_EQ(poolCalls.size(), ++pool_call_count);
-
-    ASSERT_EQ(providerCalls["alloc"], 2);
-    ASSERT_EQ(providerCalls.size(), provider_call_count);
-
-    umfPoolRealloc(tracingPool.get(), nullptr, 0);
-    ASSERT_EQ(poolCalls["realloc"], 1);
-    ASSERT_EQ(poolCalls.size(), ++pool_call_count);
-
-    ASSERT_EQ(providerCalls.size(), provider_call_count);
+    // TODO: calloc is not supported by proxy pool
```

Comment:
see my previous comment - we should get the info if calloc or realloc is supported

Reply:
What do you mean by "we should get the info"? You will get UNSUPPORTED error status here.

Reply:
Hm after thinking about it, I came to the conclusion that a simple proxy pool should not be used here to test the tracking provider, as it does not test all functions. Could we just use some "malloc-pool" here? You could just use part of the old code, but rename the pool.

Reply:
Actually, I think we can still use the proxy pool here but just don't check whether the provider was called or not. So I've restored some of the original code here. 

This test only verifies that a proper umfPool* function has been called anyway. For functional testing, we have different tests.